File: src/ClientApp/src/app/pages/roles/view-models/role-detail.view-model.ts
import { RoleUserListItemViewModel } from './role-user-list-item.view-model';

export interface RoleDetailViewModel {
    id: string;
    name: string;
    code?: string;
    description: string;
    isSystemRole: boolean;
    isActive: boolean;
    userCount: number;
    linkedUsers: RoleUserListItemViewModel[];
    createdAt: string;
    createdBy: string;
    updatedAt?: string;
    updatedBy?: string;
}

File: src/API/src/ERP.Identity/Model/ViewModels/RoleDetailViewModel.cs
namespace ERP.Identity.Model.ViewModels
{
    public class RoleDetailViewModel
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Code { get; set; }
        public string Description { get; set; } = string.Empty;
        public bool IsSystemRole { get; set; }
        public bool IsActive { get; set; }
        public int UserCount { get; set; }
        public List<RoleUserListItemViewModel> LinkedUsers { get; set; } = new();
        public DateTime CreatedAt { get; set; }
        public Guid CreatedBy { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public Guid? UpdatedBy { get; set; }
    }
}
